File: app/src/main/java/org/ar25/androidno/api/HtmlPostConverterFactory.java
package org.ar25.androidno.api;

import org.ar25.androidno.entities.Post;

import java.lang.annotation.Annotation;
import java.lang.reflect.Type;
import java.util.List;

import okhttp3.ResponseBody;
import retrofit2.Converter;
import retrofit2.Retrofit;

public class HtmlPostConverterFactory extends Converter.Factory {
  @Override public Converter<ResponseBody, ?> responseBodyConverter(Type type, Annotation[] annotations, Retrofit retrofit) {

    if(type.equals(Post.class)){
      return new HtmlResponsePostConverter();
    }

    else {
      return new HtmlResponsePostsListConverter();
    }
  }
}

File: app/src/main/java/org/ar25/androidno/api/HtmlPostConverterFactory.kt
package org.ar25.androidno.api

import org.ar25.androidno.entities.Post
import java.lang.reflect.Type

import okhttp3.ResponseBody
import retrofit2.Converter
import retrofit2.Retrofit

class HtmlPostConverterFactory : Converter.Factory() {
    override fun responseBodyConverter(type: Type?, annotations: Array<Annotation>?, retrofit: Retrofit?): Converter<ResponseBody, *> {

        if (type == Post::class.java) {
            return HtmlResponsePostConverter()
        } else {
            return HtmlResponsePostsListConverter()
        }
    }
}
